horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -59,31 +68,130 @@
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertNotNull(node);
+      assertTrue(node.isArray());
+      assertTrue(node.isEmpty());
    }
 
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutExtractors(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Schema schema = createExampleSchema(info, false);
+
+      assertNotNull(schema.id);
+      NamedJsonPath path = null; // no extractors
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, "", path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, transformer);
+      uploadRun(runWithValue(schema, 42.0d), test.name);
+
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertTrue(node.isEmpty());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithNestedSchema(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Schema acmeSchema = createExampleSchema("AcmeCorp", "AcmeInc", "AcmeRrUs", false);
+      assertNotNull(acmeSchema.id);
+      Schema roadRunnerSchema = createExampleSchema("RoadRunnerCorp", "RoadRunnerInc", "RoadRunnerRrUs", false);
+      assertNotNull(roadRunnerSchema.id);
+
+      String variable = "foo";
+      NamedJsonPath acmePath = createExampleNamedJsonPath(variable, "$.value");
+      String function = "({ "+variable+" }) => { return JSON.stringify(({  \"outcome\": { "+variable+" } })) }";
+      Transformer acmeTransformer = createExampleTransformerWithJsonPath("acme", acmeSchema, function, acmePath);
+      NamedJsonPath roadRunnerPath = createExampleNamedJsonPath("bah", "$.value");
+      variable = "bah";
+      function = "({ "+variable+" }) => { return JSON.stringify(({  \"outcome\": { "+variable+" } })) }";
+      Transformer roadRunnerTransformer = createExampleTransformerWithJsonPath("roadrunner", roadRunnerSchema, function, roadRunnerPath);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, acmeTransformer, roadRunnerTransformer);
+      int runId = uploadRun(runWithValue(42.0d, acmeSchema, roadRunnerSchema), test.name);
+
+      assertNotEquals(-1, runId);
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertEquals(4 , node.size());
+      JsonNode acme = node.path(0).path("outcome").path("foo");
+      validate( "42", acme);
+      JsonNode runner = node.path(3).path("outcome").path("bah").path(0);
+      validate( "42", runner);
+   }
```

Comment:
Please assert that there's only one Dataset in the DB here.

Reply:
ok